securitas/templates/_form_macros.html

```diff
@@ -1,24 +1,39 @@
 {% macro with_errors(field) %}
-  {% if kwargs.pop('nolabel', '') %}
-    {% set label = False %}
-  {% else %}
-    {% set label = True %}
-  {% endif %}
-  {% if field.errors %}
-    {% set css_class = 'form-control is-invalid ' + kwargs.pop('class', '') %}
-  {% else %}
-    {% set css_class = 'form-control ' + kwargs.pop('class', '') %}
-  {% endif %}
+  {% set css_class = 'form-control ' + kwargs.pop('class', '') %}
 
-  {% if label %}
-    <label for="{{ field.name }}" class="mb-0 font-weight-bold">{{ field.label }}</label>
-  {% endif %}
-  {{ field(class=css_class, **kwargs) }}
-  
-  {% if field.errors %}
-  <div class="invalid-feedback">
-      {{ field.errors|map('escape')|join(' ') }}
-  </div>
+  {% if field.type == 'FieldList' %}
+    {# append 4 empty entries at the bottom of the fieldlist #}
+    {% for i in range(4) %}
+      {% set entry = field.append_entry() %}
```

Comment:
If feels a bit strange to be calling it here in this manner, because we don't actually care about the `entry` value we set. On top of that, this will add 4 empty entries for all `FieldList` form fields and we may want to either not do that for other forms or add a different number of empty fields. I think this code should live in the controller. So i suggest removing this here and making this change:
```diff
--- a/securitas/controller/user.py
+++ b/securitas/controller/user.py
@@ -56,4 +56,8 @@ def user_edit(ipa, username):
             flash('Profile has been succesfully updated.', 'success')
             return redirect(url_for('user', username=username))
 
+    # Append 4 empty entries at the bottom of the fieldlist
+    for i in range(4):
+        form.gpgkeys.append_entry()
+
     return render_template('user-edit.html', user=user, form=form)
```

We'll probably have to do the same for SSH keys, but I do think it's better here than in a template (even though Jinja lets us do pretty advanced python inside templates).

Reply:
yeah, this is def the better approach. the only reason for `set entry` was because it was returning the feild into the HTML, which was what i didnt want since i was iterating it afterwards.